clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -232,6 +232,8 @@ void run(long now) {
                 // This is already logged as error, but propagated here to perform any clean ups.
                 log.trace("Authentication exception while processing transactional request: {}", e);
                 transactionManager.authenticationFailed(e);
+            } catch (InterruptedException e) {
+                log.warn("InterruptedException is caught: {}", e);
```

Comment:
Since the interrupt status has been cleared by using `Thread.interrupted()` and `InterruptedException` is caught here, the  outer `run` call  further up the call stack knows nothing about Sender thread had been interrupted and will keep running.  An alternative way is not to capture `InterruptedException` and let it be thrown the outer `run`.